src/scene/scene.js

```diff
@@ -372,6 +399,17 @@ Scene.prototype._updateSkybox = function (device) {
             return;
         }
 
+        // update rotation matrix after used texture is determined, but before material is updated
+        if (!this._skyboxRotationMatrix4) this._skyboxRotationMatrix4 = new Mat4();
```

Comment:
can this block of code be moved to line 435 (inside the test) since that's where the matrix is used?

Reply:
yes I think it can be moved - I will move it

Reply:
moved - thanks